p2p/transport_test.go

```diff
@@ -138,10 +138,14 @@ func TestTransportMultiplexConnFilterTimeout(t *testing.T) {
 }
 
 func TestTransportMultiplexMaxIncomingConnections(t *testing.T) {
+	pv := ed25519.GenPrivKey()
```

Comment:
Refactored test to test max number of inbound connections in transport as previous test had max set to unlimited

--- FAIL: TestTransportMultiplexMaxIncomingConnections (0.02s)
##[error]    transport_test.go:178: expected connection reset by peer error, got auth failure: secret conn failed: EOF
FAIL
FAIL	github.com/tendermint/tendermint/p2p	5.036s


I note the last changes to this test mention the committer was unconvinced of his solution. This might be the reason:

"I think in this test they are trying to test that any connection over max incoming connections is rejected. However, setting max incoming connections to 0 actually means that the number of connections is unlimited if one looks in transport.go"